packages/xo-server/src/api/vm.mjs

```diff
@@ -655,6 +656,12 @@ export const set = defer(async function ($defer, params) {
     await this.getXapiObject(VM).update_xenstore_data(mapKeys(xenStoreData, (v, k) => autoPrefix('vm-data/', k)))
   }
 
+  const creation = extract(params, 'creation')
+  if (creation !== undefined) {
+    const xapiVm = await this.getXapiObject(VM)
+    await xoData.set(xapiVm, { creation })
```

Comment:
Since `creation` only contains `user`, we need to pass `VM.creation` to `xoData.set`, in order to keep the data present in `VM.creation`.
```suggestion
    await xoData.set(xapiVm, { creation: { ...VM.creation, ...creation } })
```